### DISCUSSION TITLE
segment从存储加载到内存中，什么时候做反序列化(基于segment binlog生成segment data)？

### DISCUSSION THREAD

**Discussion by loomrain** [milvus-io/milvus#30300] [Q&A and General discussion] [answered]
这两天在看load segment的源码(基于v2.3.5)，milvus将data和index存在binlog中，在load时将binlog从存储加载到内存，调用接口是func (loader *segmentLoader) Load()，我仔细看了这个函数的实现，但是仅看到加载binlog及其相关info的逻辑，并未明显看到对这些binlog进行了什么处理。是否意味着在load阶段，只是将segment的Binlog及其相关info简单的加载到内存，在此接口(Load)中并未进行反序列化？如果要看反序列化相关的逻辑，我需要看哪里的源码，我对此非常感兴趣，期望获得解答。

**Comment by yhmo** [comment, marked as answer]
最终会调用到这个函数：https://github.com/milvus-io/milvus/blob/d87726e4c7fde4b4bf90df965f2ee51466934823/internal/core/src/storage/Util.cpp#L510

**Reply by loomrain** [reply]
感谢，已经理清楚了整个大逻辑。确实是在load时就对binlog做了反序列化。
